opentreemap/treemap/udf.py

```diff
@@ -766,111 +877,115 @@ def datatype_by_field(self):
 
         return datatypes
 
-    def reverse_clean(self, value):
-        if self.datatype_dict['type'] == 'user':
-            if hasattr(value, 'pk'):
-                value = str(value.pk)
-        elif self.datatype_dict['type'] == 'multichoice':
-            if value and len(value) > 0:
-                value = json.dumps(value, ensure_ascii=False)
-            else:
-                value = None  # so "missing data" searches will work
-        if value:
-            return value if isinstance(value, unicode) else str(value)
+    def reverse_clean(self, value, key=None, datatype_dict=None):
+        dtd = datatype_dict or self.datatype_dict
+        dtd = dtd if key is None else self.datatype_by_field[key]
+        datatype = dtd['type']
+
+        if datatype == 'date':
+            if isinstance(value, six.string_types):
+                return value
+            elif isinstance(value, datetime):
+                return value.strftime(DATETIME_FORMAT)
+            elif isinstance(value, Iterable):
+                fmt = DATE_FORMAT
```

Comment:
It used to test against `list`, but one of the tests passed in a `tuple`, which failed. It's all about getting tests to pass.

Should a comment cite specific tests?